src/main/kotlin/de/itemis/mps/gradle/generate/Plugin.kt

```diff
@@ -60,15 +66,15 @@ open class GenerateMpsProjectPlugin : Plugin<Project> {
                 *
                 */
                 val fake = tasks.create("fakeBuildNumber", FakeBuildNumberTask::class.java) {
-                    mpsDir = mpsLocation
+                    mpsDir.set(mpsLocation)
```

Comment:
if the mpsLocation property gets a convention as I suggested above then we can probably use `extension.mpsLocation` here directly.

Reply:
Moving to properties is what @coolya suggested, which also decreses built-times due to its lazy behavior. Maybe we three should have a short meeting this topic ....

Reply:
> Maybe we three should have a short meeting this topic

Probably. Also it would be better if we could split the PR, upgrade to Gradle 7.4 in this PR and leave properties for another one.

> I injected 'Project', but I do not know how to get rid of 'mpsDir.set(mpsLocation)' without changing 'FakeBuildNumberTask' quite drastically ....

I think you could then directly use `mpsDir.set(extension.mpsLocation)`.

I have tried this locally and found that adding a convention to mpsLocation means that we can no longer complain if the user sets only mpsVersion but no mpsLocation, so this changes the behavior a bit from what is documented. Not sure if we want that. If not, then your original solution without conventions is fine.